pkg/services/ngalert/eval/eval_test.go

```diff
@@ -276,6 +276,40 @@ func TestEvaluateExecutionResult(t *testing.T) {
 				},
 			},
 		},
+		{
+			desc: "no data for Ref ID will produce DatasourceNoData error",
```

Comment:
test descriptions are not very clear to me. The evaluation does not know about `DatasourceNoData`. Probably it is better to mention that NoData state provides labels datasource_uid